src/Build/Utilities/Utilities.cs

```diff
@@ -618,7 +618,7 @@ public static T[] ToArray<T>(this IEnumerator<T> enumerator)
             return enumerator.ToEnumerable().ToArray();
         }
 
-        public static void EnumerateProperties(IEnumerable properties, Action<KeyValuePair<string, string>> callback)
+        public static void EnumerateProperties<TArg>(IEnumerable properties, TArg arg, Action<TArg, KeyValuePair<string, string>> callback)
```

Comment:
I think a rename like this would have helped me understand the idea.

```suggestion
        public static void EnumerateProperties<TAccumulator>(IEnumerable properties, TAccumulator acumulator, Action<TAccumulator, KeyValuePair<string, string>> callback)
```

Reply:
`TArg` is the name commonly used for this pattern, such as in [`System.Collections.Immutable.ImmutableInterlocked`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutableinterlocked?view=net-6.0) and [`System.Collections.Concurrent.ConcurrentDictionary<TKey,TValue>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-6.0).

Reply:
My $0.02 would be that `TAccumulator` makes sense when you need the collection element type, as `TArg` refers to an arbitrary argument.

Expressed in c#, generic constraints would be expressed as `Sum<T, TArg, TAccumulator>(this T target, TArg arg, TAccumulator initial) where T:IEnumerable<TAccumulator>`

Reply:
Did we end on using TArg or TAccumulator here? I don't have much of a preference myself; just want to know if it's ready to merge or not.

Reply:
I still prefer `TAccumulator` but not enough to block the merge on it :)